privacyidea/lib/tokens/registrationtoken.py

```diff
@@ -164,7 +165,17 @@ def update(self, param):
             # Otherwise genkey and otpkey will raise an exception in
             # PasswordTokenClass
             del param["genkey"]
-        param["otpkey"] = generate_password(size=self.otp_len)
+        if "registrationcode_length" in param:
+            size = param["registrationcode_length"]
+            del param["registrationcode_length"]
+        else:
+            size = self.otp_len
+        if "registrationcode_contents" in param:
+            otpkey = _generate_pin_from_policy(param["registrationcode_contents"], size=int(size))
+            del param["registrationcode_contents"]
+        else:
+            otpkey = generate_password(size=int(size))
```

Comment:
Can't we always use _generate_pin_from_policy, and provide the size:

    otpkey = _generate_pin_from_policy(param["registrationcode_contents"], size)

Reply:
if the parameter registration.contents has proper defaults, one can, indeed. Done.